Sources/SwiftWarningControl/WarningControlRegionBuilder.swift

```diff
@@ -0,0 +1,119 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// Compute the full set of warning control regions in this syntax node
+extension SyntaxProtocol {
+  @_spi(ExperimentalLanguageFeatures)
+  public func warningGroupControlRegionTree() -> WarningControlRegionTree {
+    let visitor = WarningControlRegionVisitor(
+      start: self.position,
+      end: self.endPosition
+    )
+    visitor.walk(self)
+    return visitor.tree
+  }
+}
+
+/// Add this warning control decl syntax node warning group controls (as specified with `@warn`)
+/// to the specified WarningControlRegionTree.
+extension WarningControlDeclSyntax {
+  func addWarningControlRegionsToTree(_ tree: inout WarningControlRegionTree) {
```

Comment:
Wouldn’t it make more sense if this was a method on `WarningControlRegionTree` that takes `some WarningControlDeclSyntax` as argument?

Reply:
Changed, thanks.